clients/src/test/java/org/apache/kafka/common/security/kerberos/KerberosNameTest.java

```diff
@@ -55,4 +55,33 @@ public void testParse() throws IOException {
         assertEquals("REALM.COM", name.realm());
         assertEquals("user", shortNamer.shortName(name));
     }
+
+    @Test
+    public void testToLowerCase() throws Exception {
```

Comment:
Do we have any test cases which validate the behavior for invalid rules?

Reply:
@hachikuji  Thanks for the review. Added a simple test case for invalid rules. Pls take a look.